torch/lib/THPP/tensors/generic/THTensor.cpp

```diff
@@ -249,12 +295,251 @@ auto THTensor<real>::unsqueeze(const Tensor& src, int dimension) -> THTensor& {
 }
 
 template<>
+auto THTensor<real>::gesv(const Tensor& ra, const Tensor& b, const Tensor& a) -> THTensor& {
+#if defined(TH_REAL_IS_DOUBLE) || defined(TH_REAL_IS_FLOAT)
```

Comment:
Are Lapack functions only available for `float` and `double`? I cannot find any define guards in `TH`.